### PR TITLE
Fix usage of moved local variable in test utils

### PR DESCRIPTION
**Pull request by SergeyKopienko**
In this PR we fix usage of potentially moved object in test utils

### CODE REVIEW

File: test/support/utils_invoke.h [hunk 1]
```diff
@@ -162,7 +162,7 @@ struct invoke_on_all_policies
     void
     operator()(Op op, T&&... rest)
     {
-        invoke_on_all_host_policies()(op, ::std::forward<T>(rest)...);
+        invoke_on_all_host_policies()(op, rest...);
```
Comment (MikeDvorskiy) [inline test/support/utils_invoke.h:165]: What if `TEST_DPCPP_BACKEND_PRESENT` defined in 0?

Comment (SergeyKopienko) [reply test/support/utils_invoke.h]: Fixed.

**Review by MikeDvorskiy — DISMISSED** [review]
LGTM

**Comment by rarutyun** [comment]
From the code perspective the fix is right. From the "coverage" perspective we lose something because rvalues now have are tested only with `par_unseq` or with `bidirectional_iterator_tag` or with (what concerns me more) just with hetero policies, that reduces our test variation (the last case, could do it drastically)

I don't have immediate suggestions what is the best way to solve that. We could discuss.

**Review by rarutyun — APPROVED** [review]

**Pull request merged**